frontend/amundsen_application/static/js/features/PreviewData/index.tsx

```diff
@@ -0,0 +1,95 @@
+// Copyright Contributors to the Amundsen project.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as React from 'react';
+
+import { PreviewData } from 'interfaces/PreviewData';
+import {
+  NO_DATA_MESSAGE,
+  PREVIEW_COLUMN_MAX_LEN,
+  PREVIEW_COLUMN_MSG,
+} from './constants';
+
+import './styles.scss';
+
+interface PreviewDataProps {
+  isLoading: boolean;
+  previewData: PreviewData;
+}
+
+export const PreviewDataLoader = () => (
+  <div className="preview-data-loader">
+    <div className="shimmer-header-row is-shimmer-animated" />
+    <div className="shimmer-row is-shimmer-animated" />
+    <div className="shimmer-row is-shimmer-animated" />
+    <div className="shimmer-row is-shimmer-animated" />
+    <div className="shimmer-row is-shimmer-animated" />
+    <div className="shimmer-row is-shimmer-animated" />
+    <div className="shimmer-row is-shimmer-animated" />
+    <div className="shimmer-row is-shimmer-animated" />
+    <div className="shimmer-row is-shimmer-animated" />
+  </div>
+);
+
+const getSanitizedValue = (value) => {
+  let sanitizedValue = '';
+  if (value === 0 || typeof value === 'boolean') {
+    sanitizedValue = value.toString();
+  } else if (typeof value === 'object') {
+    sanitizedValue = JSON.stringify(value);
+  } else {
+    sanitizedValue = value;
+  }
+
+  if (sanitizedValue.length > PREVIEW_COLUMN_MAX_LEN) {
+    return PREVIEW_COLUMN_MSG;
+  }
+  return sanitizedValue;
+};
+
+export const PreviewDataTable: React.FC<PreviewDataProps> = ({
+  isLoading,
+  previewData,
+}) => {
+  if (isLoading) {
+    return <PreviewDataLoader />;
+  }
+  if (
+    !previewData.columns ||
+    !previewData.data ||
+    previewData.columns.length === 0 ||
+    previewData.data.length === 0
+  ) {
```

Comment:
Nit: I would extract a variable named hasData from this logic and use it here

Reply:
Most of this is copy-pasted from the `DataPreviewButton`. I'll clean up a bit.

Reply:
TS compiler doesn't like it when I use a variable. It complains that a usage of `previewData.columns` below might be undefined.

Reply:
Ah... sometimes happens, TS!!